File: holoken-core/src/integrationTest/java/com/holokenmod/creation/TestGridCreator.java
package com.holokenmod.creation;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

import com.holokenmod.grid.Grid;
import com.holokenmod.grid.GridSize;
import com.holokenmod.options.CurrentGameOptionsVariant;
import com.holokenmod.options.DigitSetting;
import com.holokenmod.options.GridCageOperation;
import com.holokenmod.options.SingleCageUsage;

import org.junit.jupiter.api.Test;

public class TestGridCreator {
	@Test
	void test3x3GridCreationWithoutRandomValues() {
		CurrentGameOptionsVariant.getInstance().setDigitSetting(DigitSetting.FIRST_DIGIT_ONE);
		CurrentGameOptionsVariant.getInstance().setShowOperators(true);
		CurrentGameOptionsVariant.getInstance().setSingleCageUsage(SingleCageUsage.FIXED_NUMBER);
		CurrentGameOptionsVariant.getInstance().setCageOperation(GridCageOperation.OPERATIONS_ALL);
		
		GridCreator creator = new GridCreator(
				new RandomizerMock(),
				new ShufflerStub(),
				new GridSize(3, 3));
		
		Grid grid = creator.createRandomizedGridWithCages();
		
		assertThat(grid.getCellAt(0, 0).getValue(), is(1));
		assertThat(grid.getCellAt(0, 1).getValue(), is(2));
		assertThat(grid.getCellAt(0, 2).getValue(), is(3));
		
		assertThat(grid.getCellAt(1, 0).getValue(), is(2));
		assertThat(grid.getCellAt(1, 1).getValue(), is(3));
		assertThat(grid.getCellAt(1, 2).getValue(), is(1));
		
		assertThat(grid.getCellAt(2, 0).getValue(), is(3));
		assertThat(grid.getCellAt(2, 1).getValue(), is(1));
		assertThat(grid.getCellAt(2, 2).getValue(), is(2));
	}
}

File: holoken-core/src/integrationTest/kotlin/com/holokenmod/creation/TestGridCreator.kt
package com.holokenmod.creation

import com.holokenmod.grid.GridSize
import com.holokenmod.options.DifficultySetting
import com.holokenmod.options.DigitSetting
import com.holokenmod.options.GameOptionsVariant
import com.holokenmod.options.GameVariant
import com.holokenmod.options.GridCageOperation
import com.holokenmod.options.SingleCageUsage
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class TestGridCreator : FunSpec({
    test("3x3GridCreationWithoutRandomValues") {
        val creator = GridCreator(
            RandomizerMock(),
            ShufflerStub(),
            GameVariant(
                GridSize(3, 3),
                GameOptionsVariant(
                    true,
                    GridCageOperation.OPERATIONS_ALL,
                    DigitSetting.FIRST_DIGIT_ONE,
                    DifficultySetting.ANY,
                    SingleCageUsage.FIXED_NUMBER,
                    false
                )
            )
        )

        val grid = creator.createRandomizedGridWithCages()

        grid.getValidCellAt(0, 0).value shouldBe 1
        grid.getValidCellAt(0, 1).value shouldBe 2
        grid.getValidCellAt(0, 2).value shouldBe 3
        grid.getValidCellAt(1, 0).value shouldBe 2
        grid.getValidCellAt(1, 1).value shouldBe 3
        grid.getValidCellAt(1, 2).value shouldBe 1
        grid.getValidCellAt(2, 0).value shouldBe 3
        grid.getValidCellAt(2, 1).value shouldBe 1
        grid.getValidCellAt(2, 2).value shouldBe 2
    }
})
